test/integration_test.py

```diff
@@ -1113,9 +1119,18 @@ def test_can_bundle_canned_repository(self, mock_default_catalog, mock_catalogs)
 
     def bundle_fqid(self, catalog: CatalogName) -> SourcedBundleFQID:
         plugin = RepositoryPlugin.load(catalog).create(catalog)
-        spec = first(plugin.sources)
-        source = plugin.resolve_source(str(spec))
-        return first(plugin.list_bundles(source, prefix=source.spec.prefix))
+        sources = [plugin.resolve_source(str(spec)) for spec in plugin.sources]
+        prefix = random_prefix(length=4)
+        while True:
+            bundle_fqids = chain.from_iterable(
+                plugin.list_bundles(source, prefix=prefix)
+                for source in sources
+            )
+            try:
+                return first(bundle_fqids)
```

Comment:
I'm having a hard time reasoning about the randomness of this since you are chaining the results from the sources in a deterministic order. It wouldn't hurt to pick a random element.